### DISCUSSION TITLE
RefindPlus Fails to Build with Monterey

### DISCUSSION THREAD

**Discussion by phantomenace117** [RefindPlusRepo/RefindPlusUDK#14] [General]
i have a 2008 CMP 3.1 that is currently corresponding with you via Mac OS Monterey via OCLP. Through that process, I learned a lot, and I'm now trying to install other OS on the computer, Windows 10 being the primary one. Everything I've seen points to RefindPlus being perfect for what I want to do. My issue is that after following all of the steps, to the letter, at this link. https://github.com/dakanji/RefindPlus/blob/GOPFix/BUILDING.md when i get to the build phase, it seems to be giving out more error codes than not and it ends in a "runtime error" in terminal. I'm not sure where i went wrong. I had to take a additional step of changing permissions for the RefindPlus folder (using "sudo chmod 755 'filename'") so that refindplusbuilder.sh could be launched in terminal. after doing that, it runs, but it ultimately fails. any help would be appreciated! I would like to have linux windows and modern mac os all on this machine if possible! thanks :)

**Comment by dakanji** [comment]
Can you let me know which version of Mac OS you are trying to build on?
I assume Monterey which I have never tried to build on myself and need to test if this is the case.
Please confirm.

Also, need to see where the build script is failing exactly.
Please attach a screenshot of the terminal screen.


Thanks

**Comment by phantomenace117** [comment]
Yes, I am trying Monterey booting through OCLP. If you’ve had success with
another OS, I could try that also, but I’m down to be a Guinea pig for
Monterey as well. I’ll get you a screen shot of the screen to follow this
email, it seems to be failing after completing the clean phase and when it
tries to build the toolkit it’s giving me an error message.

**Comment by phantomenace117** [comment]
hopefully these screenshots uploaded in order, i tried to include the
entire sequence for you. you can even see at the beginning where when I
build the infrastructure it requires me to force permissions at launch even
though I'm doing the operation as the super user.

**Comment by phantomenace117** [comment]
![Screen Shot 2024-01-10 at 3 56 34 AM](https://github.com/dakanji/RefindPlus/assets/156055008/e697088c-c907-4c21-95ac-64560bc97e7b)
![Screen Shot 2024-01-10 at 3 56 48 AM](https://github.com/dakanji/RefindPlus/assets/156055008/005377f9-472f-4886-a9c4-2c10878519af)
![Screen Shot 2024-01-10 at 3 57 33 AM](https://github.com/dakanji/RefindPlus/assets/156055008/222d145c-e557-4ee5-81b6-6755124426c2)
![Screen Shot 2024-01-10 at 3 57 49 AM](https://github.com/dakanji/RefindPlus/assets/156055008/843e8e39-a520-41c2-9290-f1a0ae64d37a)
![Screen Shot 2024-01-10 at 3 58 00 AM](https://github.com/dakanji/RefindPlus/assets/156055008/6e55675a-47d6-47af-b216-249a421cbb1f)

**Comment by dakanji** [comment]
Thanks.

I run Monterey on my MP31 as well but I build in BigSur (Mojave previously).
I will try to build in Monterey to see what gives and get back to you presently.

**Comment by phantomenace117** [comment]
Thanks for that insight! I’ll be patiently waiting! I appreciate your
diligence!

**Comment by dakanji** [comment]
Took a proper look at the screenshots and I can see the issue.
Python was basically removed from Monterey and certain parts of the build relies on this

Your options:
- Use the prepackaged RefindPlus from the releases tab. The build's Python dependency will be looked at later
- Follow this: https://github.com/dakanji/RefindPlusUDK/issues/12#issuecomment-1318654819
- Follow this: https://github.com/dakanji/RefindPlusUDK/issues/12#issuecomment-1318681662

**Comment by phantomenace117** [comment]
Thanks a lot! I’ll look into it as I get time today! I may step back to a
clean install of Big Sur and work from there, if I still struggle on
Monterey!

**Comment by dakanji** [comment]
Good luck.

I will move this under RefindPlusUDK

**Comment by dakanji** [comment]
Can you try changing the last line of your `Documents/RefindPlus/edk2/BaseTools/BinWrappers/PosixLike/build` file
FROM: `exec "${python_exe:-python}" "$dir/../../Source/Python/$cmd/$cmd.py" "$@"`
TO: `exec "${python_exe:-python}" "$dir/../../Source/Python/$cmd/$cmd.py" "$@" || true`

That is, add ` || true` to the end and see if it resolves the issue?

**Comment by phantomenace117** [comment]
I will try that this evening and report back! Thanks!

**Comment by phantomenace117** [comment]
Here are my results after making the change you suggested. It seems to have
completed the toolkit, but still says i have a runtime error! ive included
the screen shots from this attempt. thanks for your help!

**Comment by dakanji** [comment]
I see. Please revert the change and use one of the other options.

**Comment by phantomenace117** [comment]
I will thank you!
